File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());
        vector<vector<int>> overlapIntervals;
        overlapIntervals.push_back(intervals[0]);
        for(int i = 1; i < intervals.size(); i++){
            vector<int> overlap = overlapIntervals[overlapIntervals.size()-1];
            if(intervals[i][0] > overlap[1]){
                overlapIntervals.push_back(intervals[i]);
                continue;
            }
            overlap[1] = max(intervals[i][1], overlap[1]);
            overlapIntervals[overlapIntervals.size()-1] = overlap;
        }
        return overlapIntervals;
    }
};

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        newIntervals = []
        intervals.sort()
        currentInterval = intervals[0]
        for interval in intervals:
            if interval[0] <= currentInterval[1]:
                print(interval[0], currentInterval[1])
                currentInterval[0] = min(currentInterval[0], interval[0])
                currentInterval[1] = max(currentInterval[1], interval[1])
                print(currentInterval)
            else:
                newIntervals.append(currentInterval.copy())
                currentInterval = interval.copy()
        newIntervals.append(currentInterval)
        return newIntervals